homeassistant/components/nest/config_flow.py

```diff
@@ -240,72 +249,77 @@ async def async_step_device_project(
+        cloud_project_id = self._data[CONF_CLOUD_PROJECT_ID]
+        if self._admin_client is None:
             access_token = self._data["token"]["access_token"]
             self._admin_client = api.new_pubsub_admin_client(
-                self.hass, access_token=access_token, cloud_project_id=cloud_project_id
+                self.hass,
+                access_token=access_token,
+                cloud_project_id=cloud_project_id,
             )
-            try:
-                eligible_topics = await self._admin_client.list_eligible_topics(
-                    device_access_project_id=device_access_project_id
-                )
-            except ApiException as err:
-                _LOGGER.error("Error listing eligible Pub/Sub topics: %s", err)
-                errors["base"] = "pubsub_api_error"
-            else:
-                if not eligible_topics.topic_names:
-                    errors["base"] = "no_pubsub_topics"
+        errors = {}
+        if user_input is not None:
+            topic_name = user_input[CONF_TOPIC_NAME]
+            if topic_name == CREATE_NEW_TOPIC_KEY:
+                topic_name = _generate_topic_id(cloud_project_id)
+                _LOGGER.debug("Creating topic %s", topic_name)
+                try:
+                    await self._admin_client.create_topic(topic_name)
+                    await self._admin_client.set_topic_iam_policy(
+                        topic_name, DEFAULT_TOPIC_IAM_POLICY
+                    )
+                except ApiException as err:
+                    _LOGGER.error("Error creating Pub/Sub topic: %s", err)
+                    errors["base"] = "pubsub_api_error"
             if not errors:
-                self._data[CONF_CLOUD_PROJECT_ID] = cloud_project_id
-                self._eligible_topics = eligible_topics
-                return await self.async_step_pubsub_topic()
+                self._data[CONF_TOPIC_NAME] = topic_name
+                return await self.async_step_pubsub_topic_confirm()
+
+        device_access_project_id = self._data[CONF_PROJECT_ID]
+        try:
+            eligible_topics = await self._admin_client.list_eligible_topics(
+                device_access_project_id=device_access_project_id
+            )
+        except ApiException as err:
+            _LOGGER.error("Error listing eligible Pub/Sub topics: %s", err)
+            return self.async_abort(reason="pubsub_api_error")
+
+        topics = {topic_name: topic_name for topic_name in eligible_topics.topic_names}
+        topics[CREATE_NEW_TOPIC_KEY] = "Create New"
 
         return self.async_show_form(
-            step_id="pubsub",
+            step_id="pubsub_topic",
             data_schema=vol.Schema(
                 {
-                    vol.Required(CONF_CLOUD_PROJECT_ID, default=cloud_project_id): str,
+                    vol.Optional(CONF_TOPIC_NAME, default=next(iter(topics))): vol.In(
+                        topics
+                    ),
```

Comment:
I believe this can be a translated thing and then we can translate Create New (I know the rest is not translated, but I believe it will just fallback to default

Reply:
fixed here and the existing subscription name also.